packages/itwin/grouping-mapping-widget/src/widget/components/groupsHelpers.ts

```diff
@@ -7,13 +7,19 @@ import type { IModelConnection } from "@itwin/core-frontend";
 import type { Group } from "@itwin/insights-client";
 import { toaster } from "@itwin/itwinui-react";
 import { KeySet } from "@itwin/presentation-common";
-import type { QueryCacheItem } from "./context/GroupHilitedElementsContext";
-import { clearEmphasizedOverriddenElements, emphasizeElements, getHiliteIds, hideElements, overrideElements, zoomToElements } from "./viewerUtils";
+import type { OverlappedElementGroupPairs, OverlappedInfo, QueryCacheItem } from "./context/GroupHilitedElementsContext";
+import { clearEmphasizedOverriddenElements, clearHiddenElements, emphasizeElements, getHiliteIds, hideElements, overrideElements, zoomToElements } from "./viewerUtils";
 
 const goldenAngle = 180 * (3 - Math.sqrt(5));
 
 export const getGroupColor = function (index: number) {
-  return `hsl(${index * goldenAngle + 60}, 100%, 50%)`;
+  let hue = (index * goldenAngle * 1.5 + 60) % 360;
```

Comment:
There are a lot of magic numbers in this function. Comment all of them, explaining why a specific value is used. Also, move those magic numbers to constants with meaningful names that would make reading this function easier.

Reply:
Done